exporters/stats/signalfx/src/main/java/io/opencensus/exporter/stats/signalfx/SignalFxStatsExporter.java

```diff
@@ -0,0 +1,135 @@
+ *   ... // Do work.
+ * }
+ * </code></pre>
+ */
+public final class SignalFxStatsExporter {
+
+  public static final String SIGNALFX_TOKEN_PROPERTY = "signalfx.auth.token";
+  public static final String SIGNALFX_HOST_PROPERTY = "signalfx.ingest.url";
+  public static final String DEFAULT_SIGNALFX_ENDPOINT = "https://ingest.signalfx.com";
+
+  private static final Duration ZERO = Duration.create(0, 0);
+  private static final Duration DEFAULT_INTERVAL = Duration.create(1, 0);
+  private static final Object monitor = new Object();
+
+  private final SignalFxStatsExporterWorkerThread workerThread;
+
+  @GuardedBy("monitor")
+  private static SignalFxStatsExporter exporter = null;
+
+  private SignalFxStatsExporter(URL url, String token, Duration interval, ViewManager viewManager) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(token), "Invalid SignalFx token");
+    Preconditions.checkNotNull(interval, "Interval must not be null");
+    Preconditions.checkArgument(interval.compareTo(ZERO) > 0, "Interval duration must be positive");
+    this.workerThread =
+        new SignalFxStatsExporterWorkerThread(
+            SignalFxMetricsSenderFactory.DEFAULT, url, token, interval, viewManager);
+  }
+
+  /**
+   * Creates a SignalFx Stats exporter that gets its authentication token and the SignalFx ingest
+   * URL from system properties and uses the default reporting interval.
+   *
+   * <p>Only one SignalFx Stats exporter can be created.
+   *
+   * @throws IllegalStateException if a SignalFx exporter already exists.
+   */
+  public static void create() {
+    create(System.getProperties());
+  }
+
+  /**
+   * Creates a SignalFx Stats exporter that gets its authentication token and the SignalFx ingest
+   * URL from the given Java properties and uses the default reporting interval.
+   *
+   * <p>Only one SignalFx Stats exporter can be created.
+   *
+   * @param properties a {@code Properties} object containing the {@link #SIGNALFX_TOKEN_PROPERTY}
+   *     and {@link #SIGNALFX_HOST_PROPERTY}.
+   * @throws IllegalStateException if a SignalFx exporter already exists.
+   */
+  public static void create(Properties properties) {
+    Preconditions.checkNotNull(properties, "Given properties shouldn't be null");
+    String host = properties.getProperty(SIGNALFX_HOST_PROPERTY, DEFAULT_SIGNALFX_ENDPOINT);
+    String token = properties.getProperty(SIGNALFX_TOKEN_PROPERTY);
+    create(host, token, DEFAULT_INTERVAL);
+  }
+
+  /**
+   * Creates a SignalFx Stats exporter with an explicit authentication token and reporting interval.
+   *
+   * <p>Only one SignalFx Stats exporter can be created.
+   *
+   * @param host the SignalFx ingest URL.
+   * @param token a SignalFx ingest token.
+   * @param interval the interval at which stats are reported to SignalFx.
+   * @throws IllegalStateException if a SignalFx exporter already exists.
+   */
+  public static void create(String host, String token, Duration interval) {
```

Comment:
Suggestion: We just switched to using a configuration class with a builder in the StackDriver exporter (#896).  That design might work well here.

Reply:
I'll look into this. The configuration here seems a lot simpler than for the StackDriver exporter, there really only is the token to specify, and the simplest way to do that is via Java system properties which can be wired through by the orchestration/deployment system (passing in a `-Dsignalfx.auth.token=`).